### DISCUSSION TITLE
Right config for RepeatNet model with yoochoose-clicks.inter

### DISCUSSION THREAD

**Discussion by Jeriousman** [RUCAIBox/RecBole#1218] [Q&A] [answered]
I am trying to test RepeatNet in the sequential recommender with yoochoose-clicks.inter data. Yoochoose-clicks.inter has features as ```['session_id', 'item_id', 'count', 'timestamp']``` which does not have user_id which seems to be mandatory. How should I set up config for RepeatNet? I set some myself but I let the rest be the default. I kinda wished using ```config = Config(model='RepeatNet', dataset = 'yoochoose-clicks', config_dict= config_dict)``` sets right config except ones I put in config_dict will change some config according to that. But after this, still config gives me an error as below
```

  File "/home/hojun/anaconda3/envs/recbole/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 3363, in get_loc
    raise KeyError(key) from err

KeyError: 'user_id'
```
It is because the config is as followed

```
General Hyper Parameters:
gpu_id = 0
use_gpu = True
seed = 2020
state = INFO
reproducibility = True
data_path = /home/hojun/Python/Models/Recommender_systems/RecBole/run_example/dataset/yoochoose/merged/yoochoose-clicks
checkpoint_dir = saved
show_progress = True
save_dataset = True
dataset_save_path = None
save_dataloaders = True
dataloaders_save_path = None
log_wandb = False

Training Hyper Parameters:
epochs = 300
train_batch_size = 2048
learner = adam
learning_rate = 0.001
neg_sampling = {'uniform': 1}
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4

Evaluation Hyper Parameters:
eval_args = {'split': {'LS': 'valid_and_test'}, 'order': 'TO', 'mode': 'full', 'group_by': 'user'}
repeatable = True
metrics = ['Recall', 'MRR', 'NDCG', 'Hit', 'Precision']
topk = [10]
valid_metric = MRR@10
valid_metric_bigger = True
eval_batch_size = 4096
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['session_id', 'item_id', 'count', 'timestamp']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = None
filter_inter_by_user_or_item = True
user_inter_num_interval = [0,inf)
item_inter_num_interval = [0,inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = None
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id
benchmark_filename = None

Other Hyper Parameters: 
wandb_project = recbole
require_pow = False
embedding_size = 64
loss_type = CE
hidden_size = 64
joint_train = False
dropout_prob = 0.2
MODEL_TYPE = ModelType.SEQUENTIAL
MODEL_INPUT_TYPE = InputType.POINTWISE
eval_type = EvaluatorType.RANKING
device = cuda
train_neg_sample_args = {'strategy': 'by', 'by': 1, 'distribution': 'uniform', 'dynamic': 'none'}
eval_neg_sample_args = {'strategy': 'full', 'distribution': 'uniform'}
```
as you can see 'USER_ID_FIELD = user_id' by default which I didnt put in my config_dict. So my question is as below:

1. What can I do about USER_ID_FIELD in yoochoose-clicks.inter? Should I put session_id as USER_ID_FIELD?

2. What is right default config for RepeatNet with yoochoose-clicks.inter in general?

3. Is there any page I can get or see safe overall default config settings of each and every different model separately? If not, any tips for the config? Because the config is huge as you are dealing with many different models.

Thank you in advance.

**Comment by hyp1231** [comment, marked as answer]
Hi,

1. Yes! you can just set `USER_ID_FIELD` as `session_id`. As yoochoose is a session-based dataset, it originally have no users in the dataset. You can regard each session-id as an (anonymous) user.
2. Try the suggestions in 1. and maybe it can run successfully.
3. It's a really good suggestion! But as you see, the combination is really huge. We have also noticed that but still have not figured out a proper way to handle this. 😮

**Reply by Jeriousman** [reply]
I will follow your suggestion. Thank you for the advice!
